src/werkzeug/exceptions.py

```diff
@@ -600,11 +600,29 @@ class TooManyRequests(HTTPException):
     to identify users and their request rates). The server may include a
     "Retry-After" header to indicate how long the user should wait before
     retrying.
+
+    .. versionchanged:: 0.16.1
+        ``retry_after_secs`` was added as the first argument, ahead of
+        ``description``.
     """
 
     code = 429
     description = "This user has exceeded an allotted request count. Try again later."
 
+    def __init__(self, description=None, retry_after_secs=None):
```

Comment:
I think it would make sense to follow `Unauthorized` and use 
```python
def __init__(self, description=None, response=None, retry_after_secs=None):
``` 
To allow a user to set the response if they want.